test-utils/test-utils-domain/src/main/java/uk/gov/justice/services/domain/main/GenericStepDefs.java

```diff
@@ -0,0 +1,213 @@
+import cucumber.api.java.en.When;
+import org.apache.commons.lang3.StringUtils;
+import org.reflections.Reflections;
+import uk.gov.justice.domain.aggregate.Aggregate;
+import uk.gov.justice.domain.annotation.Event;
+import uk.gov.justice.services.common.converter.JsonObjectToObjectConverter;
+import uk.gov.justice.services.common.converter.jackson.ObjectMapperProducer;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class GenericStepDefs {
+    private static final String METADATA = "_metadata";
+    private static final String NAME = "name";
+    private List givenEvents = new ArrayList();
+    private List generatedEvents = new ArrayList();
+    private Class<?> clazz;
+    private Aggregate object;
+    @Inject
+    private JsonObjectToObjectConverter jsonObjectToObjectConverter;
+
+    @Given("no previous events")
+    public void no_previous_events() throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+        assert givenEvents.size() == 0;
+    }
+
+    @Given("there are previous events (.*)")
+    public void previous_events(final String fileNames) throws Exception {
+        String filesNames[] = fileNames.split(",");
+        for (String fileName : filesNames) {
+            String message = json(fileName);
+            ObjectMapper mapper = new ObjectMapperProducer().objectMapper();
+            final JsonNode fromJson = mapper.readTree(message);
+            Reflections reflections = new Reflections("uk.gov");
+            Set<Class<?>> annotated = reflections.getTypesAnnotatedWith(Event.class);
+            for (Class<?> clazz : annotated) {
+                if (clazz.getAnnotation(Event.class).value().equalsIgnoreCase(eventName(fromJson))) {
+                    removeMetaDataNode(fromJson);
+                    generatedEvents.add(mapper.readValue(mapper.writeValueAsString(fromJson), clazz));
+                    break;
+                }
+            }
+        }
+    }
+
+    @When("(.*) to a (.*) using (.*)")
+    public void call_method_with_params(final String methodName, final String aggregate, final String fileName)
+            throws Exception {
+        createAggregate(aggregate);
+        ObjectMapper mapper = new ObjectMapper();//normal mapper to get the map value in correct order
+        Map argumentsMap = mapper.convertValue(mapper.readTree(json(fileName)), Map.class);
+        mapper = new ObjectMapperProducer().objectMapper();
+        List valuesList = new ArrayList(argumentsMap.values());
+        checkIfUUID(valuesList);
```

Comment:
This needs to work for all parameter types by using Jackson to convert into the target parameter type for each parameter.